### PR TITLE
fix: dashboard enhancements

### PR DESCRIPTION
**Pull request by ktx-keshavi**
- #436 
- #452 
- #424 
- #453 
- #434 & performance optimizations for chart rendering
- Panel name add on top of Add Panel page
- Show add panel button when there are no panels in a new dashboard.

### CODE REVIEW

File: web/src/components/dashboards/addPanel/ChartRender.vue [hunk 1]
```diff
@@ -167,6 +167,42 @@ export default defineComponent({
...
+    //It is used for showing long label truncate with "..."
+    const textformat =  function(layout: any){
+      console.log("layout=", layout);
+      
+      let data = layout.map((text:any)=>{
+        if(text){
+          return text.toString().length > 15 ? text.toString().substring(0, 15) + "..." : text;
+        }else {
+          return text
+        }
```
Comment (vvv) [inline web/src/components/dashboards/addPanel/ChartRender.vue:179]:
```suggestion
        if(text && text.toString().length > 15){
          return text.toString().substring(0, 15) + "...";
        } else {
          return text;
        }
```

File: web/src/components/dashboards/addPanel/ChartRender.vue [hunk 2]
```diff
@@ -196,30 +232,26 @@ export default defineComponent({
...
-          searchQueryData.data = props.data.fields?.x && props.data.fields?.x.length > 0 ? res.data.hits.sort(sortFn) : res.data.hits;
+          // const sortFn = (it1:any, it2: any) => {
+          //   const a = it1[props.data.fields?.x[0].alias] || ""
+          //   const b = it2[props.data.fields?.x[0].alias] || ""
+          //   if(typeof a == 'number' && typeof b == 'number') {
+          //       return a - b
+          //   } else {
+          //       return a.toString().localeCompare(b.toString())
+          //   }
+          // }
```
Comment (vvv) [inline web/src/components/dashboards/addPanel/ChartRender.vue:251]: Dead code. ☠️
```suggestion
```

File: web/src/components/dashboards/addPanel/IndexList.vue [hunk 3]
```diff
@@ -229,21 +257,11 @@ export default defineComponent({
         "",
         true
       ).then((res) => {
+        console.log("---list---",res.data.list);
+
```
Comment (vvv) [inline web/src/components/dashboards/addPanel/IndexList.vue:261]: This looks like a debugging output. Are you sure it is needed in users' browsers?

File: web/src/components/dashboards/addPanel/Layout.vue [hunk 4]
```diff
@@ -276,14 +276,14 @@
             label="Show Legends"
           />
           <q-form ref="" class="q-pa-sm">
-            <q-input
+            <!-- <q-input
```
Comment (vvv) [inline web/src/components/dashboards/addPanel/Layout.vue:279]: Dead code. Please remove.

File: web/src/components/dashboards/SearchBar.vue [hunk 5]
```diff
@@ -143,43 +143,59 @@ export default defineComponent({
...
+              
             }
             return selectFilter
         })
         // console.log("query: filterData",filterData);
-        
-        query += filterData.filter((it: any)=> it).join(" AND ")
+        const filterItems = filterData.filter((it: any)=> it)
+        if(filterItems.length > 0) {
+          query += "WHERE "
```
[the remark this answers is not in the document]
Comment (ktx-keshavi) [reply web/src/components/dashboards/SearchBar.vue]: Yes done

File: web/src/components/dashboards/addPanel/IndexList.vue (hunk 3 above)
Comment (ktx-keshavi) [reply web/src/components/dashboards/addPanel/IndexList.vue]: Yes removed

File: web/src/components/dashboards/addPanel/Layout.vue (hunk 4 above)
Comment (ktx-keshavi) [reply web/src/components/dashboards/addPanel/Layout.vue]: Yes removed

**Review by hengfeiyang — APPROVED** [review]

**Review by vvv — APPROVED** [review]

**Pull request merged**